seed/seed20.go

```diff
@@ -429,7 +436,24 @@ type errorComponentNotInSeed struct {
 	error
 }
 
-func (s *seed20) lookupVerifiedComponent(cref naming.ComponentRef, snapRev snap.Revision, snapID, snapProvenance string, snapsDir string, handler ContainerHandler, tm timings.Measurer) (Component, error) {
+func modelContainsComponent(modelSnaps map[string]*asserts.ModelSnap, cref naming.ComponentRef) bool {
+	sn, ok := modelSnaps[cref.SnapName]
+	if !ok {
+		return false
+	}
+
+	_, ok = sn.Components[cref.ComponentName]
+	return ok
+}
+
+func assertedComponentDir(systemDir string, modelSnaps map[string]*asserts.ModelSnap, cref naming.ComponentRef) string {
+	if modelContainsComponent(modelSnaps, cref) {
+		return filepath.Join(systemDir, "../../snaps")
```

Comment:
Maybe seed20 should have a seedDir field instead, it can be easily provided when created.

Reply:
Made this change in f1aecb548eb9e14a4f99eb5ae53370546d931a24.